packages/mgt-chat/src/statefulClient/GraphNotificationUserClient.ts

```diff
@@ -210,109 +211,128 @@ export class GraphNotificationUserClient {
+            log(`Successfully renewed user subscription ${subscription.id!}.`);
           }
+        } catch (e) {
+          error(`Failed to renew user subscription ${subscription.id!}.`, e);
+          await this.deleteCachedSubscriptions(currentUserId);
+          subscription = undefined;
         }
       }
 
+      // if there is no subscription, try to create one
       if (!subscription) {
         try {
-          subscription = await this.createSubscription(this.currentUserId);
+          this.trySwitchToDisconnected();
+          subscription = await this.createSubscription(currentUserId);
         } catch (e) {
-          subscription = undefined;
-          isRenewalInError = true;
-          error(e);
-
-          // rather than 3 seconds, back off to 10 seconds if we get a 403
-          if ((e as { statusCode?: number }).statusCode === 403) {
-            nextRenewalTimeInMs = 10 * 1000;
+          const err = e as { statusCode?: number; message: string };
+          if (err.statusCode === 403 && err.message.indexOf('has reached its limit') > 0) {
+            // if the limit is reached, back-off (NOTE: this should probably be a 429)
+            nextRenewalTimeInSec = appSettings.renewalTimerInterval * 3;
+            throw new Error(
+              `Failed to create a new subscription due to a limitation; retrying in ${nextRenewalTimeInSec} seconds: ${err.message}.`
+            );
+          } else if (err.statusCode === 403) {
+            // if true 403, stop renewal
+            error('Failed to create a new subscription due to a permanent condition; stopping renewals.', e);
+            return; // exit without setting the next renewal timer
+          } else {
+            // transient error, retry
+            throw new Error(
+              `Failed to create a new subscription due to a transient condition; retrying in ${nextRenewalTimeInSec} seconds: ${err.message}.`
+            );
           }
         }
       }
 
+      // create or reconnect the SignalR connection
       // notificationUrl comes in the form of websockets:https://graph.microsoft.com/beta/subscriptions/notificationChannel/websockets/<Id>?groupid=<UserId>&sessionid=default
       // if <Id> changes, we need to create a new connection
-      if (
-        subscription &&
-        (!this.connection ||
-          this.connection.state !== HubConnectionState.Connected ||
-          this.lastNotificationUrl !== subscription.notificationUrl)
-      ) {
-        if (subscription.notificationUrl) {
-          this.lastNotificationUrl = subscription.notificationUrl;
-          await this.createSignalRConnection(subscription.notificationUrl);
-        } else {
-          // this should never happen.
-          throw new Error('Subscription notificationUrl is invalid.');
-        }
+      if (this.connection?.state === HubConnectionState.Connected) {
+        await this.connection?.send('ping'); // ensure the connection is still alive
```

Comment:
At this point `this.connection` can be undefined.

`ping` could be anything that isn't handled or is handled in a benign way. You could name it `giraffeMonster5000` if you wanted to ensure it definitely wasn't handled. Alternatively, I chose `ping` because if it was handled the implementation should be non-problematic.